src/nomad_parser_vasp/parsers/xml_parser.py

```diff
@@ -23,83 +23,88 @@
 )
 # ? compilation mode?
-Program.compilation_host.m_annotations = dict(
-    xml=MappingAnnotationModel(path='modeling.generator.i[@name="platform"]')
+Simulation.program.compilation_host.m_annotations = dict(
+    xml=MappingAnnotationModel(path='modeling.generator.i[?"@name"="platform"]')
 )
-Simulation.m_annotations = dict(xml=MappingAnnotationModel(path='modeling'))
 KMesh.grid.m_annotations = dict(
-    xml=MappingAnnotationModel(path='modeling.kpoints.generation.v[@name="divisions"]')
+    xml=MappingAnnotationModel(
+        path='modeling.kpoints.generation.v[?"@name"="divisions"]'
+    )
 )
 KMesh.offset.m_annotations = dict(
-    xml=MappingAnnotationModel(path='modeling.kpoints.generation.v[@name="shift"]')
+    xml=MappingAnnotationModel(path='modeling.kpoints.generation.v[?"@name"="shift"]')
 )
 KMesh.high_symmetry_points.m_annotations = dict(
-    xml=MappingAnnotationModel(path='modeling.kpoints.varray.v[@name="kpointlist"]')
+    xml=MappingAnnotationModel(path='modeling.kpoints.varray.v[?"@name"="kpointlist"]')
 )
 KMesh.weights.m_annotations = dict(
-    xml=MappingAnnotationModel(path='modeling.kpoints.varray.v[@name="weights"]')
-)
-dft_path = 'modeling.calculation[@name="electronic"]'
-DFT.xc_functionals.libxc_name.m_annotations = dict(
-    xml=MappingAnnotationModel(
-        operator=(convert_xc, [dft_path + '.i[@name="GGA"]'])  # add LDA & mGGA
-    )
+    xml=MappingAnnotationModel(path='modeling.kpoints.varray.v[?"@name"="weights"]')
 )
+dft_path = 'modeling.separator[?"@name"="electronic exchange-correlation"]'
+# DFT.xc_functionals.m_annotations = dict(xml=MappingAnnotationModel(path=dft_path))
 DFT.exact_exchange_mixing_factor.m_annotations = dict(
     xml=MappingAnnotationModel(
         operator=(
             lambda mix, cond: mix if cond else 0,
-            [dft_path + '.i[@name="HFALPHA"]', dft_path + '.i[@name="LHFCALC"]'],
+            [dft_path + '.i[?"@name"="HFALPHA"]', dft_path + '.i[?"@name"="LHFCALC"]'],
         )
     )  # TODO convert vasp bool
 )
+XCFunctional.libxc_name.m_annotations = dict(
+    xml=MappingAnnotationModel(
+        operator=(convert_xc, [dft_path + '.i[?"@name"="GGA"]'])  # add LDA & mGGA
+    )
+)
 # ? target <structure name="initialpos" > and <structure name="finalpos" >
 AtomicCell.positions.m_annotations = dict(
-    xml=MappingAnnotationModel(path='calculation.structure.varray[@name="positions"]')
+    xml=MappingAnnotationModel(
+        path='calculation.structure.varray[?"@name"="positions"]'
+    )
 )
 """
 forces.m_annotations = dict(
-    xml=MappingAnnotationModel(path='calculation.structure.varray[@name="forces"]')
+    xml=MappingAnnotationModel(path='calculation.structure.varray[?"@name"="forces"]')
 )
 stress.m_annotations = dict(
-    xml=MappingAnnotationModel(path='calculation.structure.varray[@name="stress"]')
+    xml=MappingAnnotationModel(path='calculation.structure.varray[?"@name"="stress"]')
 )
 """
 AtomicCell.lattice_vectors.m_annotations = dict(
```

Comment:
i think calculation is  repeating xml section so this will not work unless you provide annotatio. to the metainfo section that correpond to calculatio

Reply:
I can invent my own section, right? Just to reference. Does not have to be of our schema.